openrtb_ext/request_wrapper.go

```diff
@@ -0,0 +1,773 @@
+package openrtb_ext
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+)
+
+// RequestWrapper wraps the OpenRTB request to provide a storage location for unmarshalled ext fields, so they
+// will not need to be unmarshalled multiple times.
+type RequestWrapper struct {
+	// json json.RawMessage
+	Request *openrtb2.BidRequest
```

Comment:
I think the JSON serialization will be the downfall of this idea. The JSON library will attempt to add the ext objects to the request, which will be a mess. The way to get around that is to declare a new `openrtb2.BidRequest` object, and then copy every first level object into that new structure. And even without serialization, at the end of the day we want to pass the adapters an `openrtb2,BidRequest` object, so we would have to go through this same process anyway in order to supply the needed object.

Reply:
We could still serialize only the embedded request, it's just the calling semantics become a bit strange. It would be along the lines of `json.Marshal(rw.BidRequest)`. This is not inheritance, just the illusion of it. You can still directly access the embedded type with the type name,

I guess it's a tradeoff between field access semantics or json marshal semantics. Since the later is used less frequently, it might be worth exploring.